File: mobile/src/main/java/com/ataulm/basic/MyViewHolder.java
package com.ataulm.basic;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

class MyViewHolder extends RecyclerView.ViewHolder {

    private final Listener listener;

    public MyViewHolder(View itemView, Listener listener) {
        super(itemView);
        this.listener = listener;
    }

    public static MyViewHolder createFrom(ViewGroup parent, Listener listener) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_my, parent, false);
        return new MyViewHolder(view, listener);
    }

    public void bind(final Item item) {
        ((TextView) itemView.findViewById(R.id.title)).setText("item " + item.id);

        ImageView heart = (ImageView) itemView.findViewById(R.id.fave);
        heart.setImageResource(item.isFaved ? R.drawable.heart_fill : R.drawable.heart_empty);

        heart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listener.onClick(item);
            }
        });
    }

    public interface Listener {

        void onClick(Item item);

    }

}

File: app/src/main/java/com/example/MyViewHolder.kt
package com.example

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.view_movie.view.*

class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    companion object {

        fun inflate(parent: ViewGroup): MyViewHolder {
            val inflater = LayoutInflater.from(parent.context)
            val itemView = inflater.inflate(R.layout.view_movie, parent, false)
            return MyViewHolder(itemView)
        }
    }

    fun bind(movie: Movie) {
        itemView.movieTitleView.text = movie.title
        itemView.movieOverviewView.text = movie.overview
//        TODO bind image
    }
}
